### ISSUE TITLE
Metasploit RPC server

### ISSUE THREAD

**Issue by WolfServer** [cSploit/android#396]
A quick suggestion, could you add an option for having the metasploit server start automatically on startup so that when I open csploit that metasploit is already ready rather than me having to wait for it to start?

**Comment by Rubenoo** [comment]
Well it doesn't work like that @wolfserver. 
Csploit does automatically start msf rpcd at startup. But it takes time before the service is fully started. This waiting is the only possible way to make msf rpcd work

**Comment by WolfServer** [comment]
I meant on system startup. Not app startup

**Comment by pgolecha** [comment]
Well @Rubenoo it is not an essential service that needs to be running everytime.
People would not want unnecessary process running in background for an app they only use once in a while

**Comment by pgolecha** [comment]
Sorry not @Rubenoo 
@wolfserver

**Comment by h439098** [comment]
it can be some option in settings. like "Start Metasploit at system start". can someone do this ?

**Comment by tux-mind** [comment]
let's make a poll and see.
how many will find it useful ?

very low priority however, lot of things to do.

**Comment by uranium-waffles** [comment]
I think it would be useful, but not necessary.

**Comment by h439098** [comment]
How it can be not necessary. I think its really good idea, because on my smartphone I need to wait Metasploit for >10 min. Its awful. And with this option in the future we not need to wait so long every time then app started. SO HOW IT CAN BE NOT NECESSARY ?

**Comment by uranium-waffles** [comment]
@h439098 Ok, calm down. For most people it only takes around 2 minutes. Like pgolecha said most people do not want an unnecessary process running for an app that is not used much. Now you could add something that says start metasploit at system start but I suggest you use android studio and make it, then compile it yourself, test it, and create a pull request because this is low priority.

**Comment by gainan** [comment]
In my opinion we could add an option and set it to false by default.

**Comment by fat-tire** [comment]
I agree-- this is almost trivial to add, but possible side-effects of having it on might be less memory available for other apps, potential slowdown of the rest of the boot (because it's loading metasploit at the same time), etc.  I wouldn't want this happening by default, but if there's call for it, (and you understand that simply starting the app at boot would kinda accompilsh the same thing) why not?

**Comment by newbie442** [comment]
For me, this would never be used.

1. Most times I start up with no expectation of using csploit - so this would generally simply slow down the start, and consume battery. 

2. Making it a selectable option makes no sense to me, as one never knows if he'll want it in a hurry the next time he starts.

3. When I do choose to use csploit, it is likely after a quick scan of the area using Fing - a lightweight scanner. At this point, I can decide if it is time to "bring up the big gun", and waiting for csploit is not an issue.

Sorry - this seems like unproductive complexity to me; developers could better invest their time on pentesting options.
